kapitan/targets.py

```diff
@@ -126,31 +128,43 @@ def compile_targets(
             # reset inventory cache and load target objs to check for missing classes
             cached.reset_inv()
             target_objs = load_target_inventory(inventory_path, updated_targets, ignore_class_notfound=False)
-        # fetch dependencies
-        if fetch:
+
             fetch_dependencies(output_path, target_objs, dep_cache_dir, force_fetch, pool)
-        # fetch targets which have force_fetch: true
-        elif not kwargs.get("force_fetch", False):
+
+        # otherwise only fetch dependencies which have force_fetch: true
+        else:
+            # ignore_class_notfound = False by default
+            target_objs = load_target_inventory(inventory_path, updated_targets)
+
             fetch_objs = []
             # iterate through targets
             for target in target_objs:
-                try:
-                    # get value of "force_fetch" property
-                    dependencies = target["dependencies"]
-                    # dependencies is still a list
-                    for entry in dependencies:
-                        force_fetch = entry["force_fetch"]
-                        if force_fetch:
-                            fetch_objs.append(target)
-                except KeyError:
-                    # targets may have no "dependencies" or "force_fetch" key
+                # get value of "force_fetch" property
+                dependencies = target.get("dependencies")
+                if not dependencies:
                     continue
-            # fetch dependencies from targets with force_fetch set to true
+
+                for dep_entry in dependencies:
+                    if not dep_entry.get("force_fetch", False):
+                        # if no fetch is needed remove the dependency and don't fetch it
+                        target["dependencies"].remove(dep_entry)
+
+                # fetch dependencies with force_fetch set to true
+                if target.get("dependencies", []):
+                    fetch_objs.append(target)
```

Comment:
idk why I reversed the logic here: so if it hasn't the flag, then we remove it...
Should I fix that?

Reply:
I think it makes sense to have a way to force on the command line "not to force" in case you are running from a place with no external connectivity

So i'd probably leave it in as it is, we can change it later if it doesn't make sense.

Reply:
Absolutely, but I think it would work the same if we say, that: if `force_fetch: true` --> force_fetch, instead of removing it from the list, if it don't have `force_fetch: true`

Reply:
> a way to force on the command line "not to force"
So you want a `--dont-force-fetch` flag?

Reply:
perhaps `--offline-mode` so it can be a generic way to turn off all attempts to download things?

@ramaro what to you think?

I am thinking of the case where you run it in places with no connectivity or you just don't want to change anything. It would suck if in those cases it kept failing.

Alternatively, we could handle it with an additional:

`fail_on_error: false`